packages/suite/src/views/wallet/trading/common/TradingForm/TradingFormOfferOTC.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from 'react';
+
+import { TradingTradeBuySellType, invityAPI } from '@suite-common/trading';
+import { Banner, Text } from '@trezor/components';
+import { spacings } from '@trezor/theme';
+
+import { Translation, TrezorLink } from 'src/components/suite';
+import { useTradingFormContext } from 'src/hooks/wallet/trading/form/useTradingCommonForm';
+import { isTradingBuyContext } from 'src/utils/wallet/trading/tradingTypingUtils';
+
+export const TradingFormOfferOTC = () => {
+    const context = useTradingFormContext<TradingTradeBuySellType>();
+    const { amountInCrypto } = context.getValues();
+    const fiatAmount = isTradingBuyContext(context)
+        ? context.getValues().fiatInput
+        : context.getValues().outputs[0].fiat;
+    const currencySelect = isTradingBuyContext(context)
+        ? context.getValues().currencySelect.value
+        : context.getValues().outputs[0].currency.value;
+    const [otcUrl, setOtcUrl] = useState<string | null>(null);
+
+    // constants will be refactor to be more dynamic in the future
+    const ALLOWED_FIAT_CURRENCIES = ['usd', 'eur'];
+    const FIAT_AMOUNT_ALLOWANCE = '50001';
+
+    const isCurrencyAllowed = ALLOWED_FIAT_CURRENCIES.includes(currencySelect);
+    const shouldBeComponentHidden =
+        amountInCrypto ||
+        !isCurrencyAllowed ||
+        !fiatAmount ||
+        Number(fiatAmount) < Number(FIAT_AMOUNT_ALLOWANCE);
+
+    useEffect(() => {
+        if (shouldBeComponentHidden) return;
+
+        const getOtcData = async () => {
+            const otcData = await invityAPI.getOTCData();
+
+            if (!otcData) return;
+
+            const apiUrl = new URL(otcData.apiUrl);
+            const params = new URLSearchParams({
+                widget_id: otcData.idWidget,
+                otc_id: otcData.idOtcUser,
+                fiat_amount: fiatAmount,
+                fiat_currency: currencySelect,
+                type: context.type,
+            });
+
+            apiUrl.search = params.toString();
+
+            setOtcUrl(apiUrl.toString());
```

Comment:
Idea: Can we get the OTC URL on click and not on render? That means when user clicks on the link we call API request?